### ISSUE TITLE
Label smoothing for SoftmaxOutput / cross-entropy loss

### ISSUE THREAD

**Issue by fhieber** [apache/incubator-mxnet#6996]
Label smoothing is a nice technique to regularize models trained with cross-entropy error, for example described [here](https://arxiv.org/pdf/1701.06548.pdf), and applied for vision [here](https://www.robots.ox.ac.uk/~vgg/rg/papers/reinception.pdf), and for NLP/MT recently [here](https://arxiv.org/abs/1706.03762).

This can be done in current MXNet when passed an explicit smoothed one-hot label distribution ([see here](https://github.com/awslabs/sockeye/blob/master/sockeye/loss.py#L101)) but this is probably wasteful w.r.t to memory if you have a large label space (e.g. large output vocabulary for LMs, translation models).
It also requires taking explicit care of padding symbols for variable-length models.

It would be nice to have such a flag in the SoftmaxOutput operator to do the smoothing internally without adding another node to the computation graph.

**Comment by zihaolucky** [comment]
follow the discussion.

**Comment by tqchen** [comment]
close as it is enabled
